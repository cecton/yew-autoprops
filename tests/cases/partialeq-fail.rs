use yew::prelude::*;
use yew_autoprops::autoprops_component;

struct TestPEqStruct<T> {
    t: T,
}

impl<T: PartialEq> PartialEq for TestPEqStruct<T> {
    fn eq(&self, other: &Self) -> bool {
        self.t == other.t
    }
}

struct NotPartialEq();

#[autoprops_component]
fn TestComponent(test_struct: &TestPEqStruct<NotPartialEq>) -> Html {
    html! {
        <div>
            { "TestComponent" }
        </div>
    }
}

fn main() {
    html! {
        <TestComponent test_struct={TestPEqStruct { t: NotPartialEq() }} />
    };
}
