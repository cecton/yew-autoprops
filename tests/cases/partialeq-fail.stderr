error[E0369]: binary operation `==` cannot be applied to type `TestPEqStruct<NotPartialEq>`
  --> tests/cases/partialeq-fail.rs:16:1
   |
16 | #[autoprops_component]
   | ^^^^^^^^^^^^^^^^^^^^^^
   |
note: an implementation of `PartialEq` might be missing for `NotPartialEq`
  --> tests/cases/partialeq-fail.rs:14:1
   |
14 | struct NotPartialEq();
   | ^^^^^^^^^^^^^^^^^^^ must implement `PartialEq`
   = note: this error originates in the derive macro `PartialEq` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `NotPartialEq` with `#[derive(PartialEq)]`
   |
14 | #[derive(PartialEq)]
   |
